packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -134,11 +134,25 @@ export class Bootstrapper {
         params.createCustomerMasterKey === false || currentKmsKeyId === undefined ? USE_AWS_MANAGED_KEY :
           undefined);
 
+    let kmsEcrKeyId = undefined;
+
+    if (params.ecrKey) {
```

Comment:
[nitpick] Why not start the `kmsEcrKeyId` with `USE_DEFAULT_KEY` and ditch one of the `else if`s?